### ISSUE TITLE
Bundle `nlua` with rocks.nvim

### ISSUE THREAD

**Issue by mrcjkb** [nvim-neorocks/rocks.nvim#386]
We already bundle a luarocks config. 
With this, we can set `variables.LUA = <path-to-nlua>`.

**Comment by mrcjkb** [comment]
Closing as not planned for now.

**Comment by mrcjkb** [comment]
Reopening. We might not be able to use nlua, as the luarocks wrapper has a hardcoded shebang, but we might be able to use a simple shim.

**Comment by bnwa** [comment]
Just to be clear, would this remove the need for users to maintain a ‘lua’ executable on path? I ask b/c on macOS, Homebrew does not install an executable with that name for any of its Lua runtime packages, so users have to maintain an alias it would seem.

**Comment by mrcjkb** [comment]
> Just to be clear, would this remove the need for users to maintain a ‘lua’ executable on path? I ask b/c on macOS, Homebrew does not install an executable with that name for any of its Lua runtime packages, so users have to maintain an alias it would seem.

Yes, this issue exists specifically because of the homebrew issue :smile:

**Comment by bnwa** [comment]
> > Just to be clear, would this remove the need for users to maintain a ‘lua’ executable on path? I ask b/c on macOS, Homebrew does not install an executable with that name for any of its Lua runtime packages, so users have to maintain an alias it would seem.
> 
> Yes, this issue exists specifically because of the homebrew issue 😄

Ohhhhh

**Comment by mrcjkb** [comment]
`rocks` is getting quite close to feature parity with luarocks (at least for the purpose of installing the vast majority of nvim plugins).
And this issue is more complicated than I anticipated. As there are known workarounds to getting this to work on macos, I will de-prioritise this. We should, however, add a wiki page with suggestions for macos.
